docs/iris/src/further_topics/metadata.rst

```diff
@@ -0,0 +1,984 @@
+
+We can use the `namedtuple._make`_ method to create a **new**
+:class:`~iris.common.DimCoordMetadata` instance from an existing sequence
+or iterable:
+
+    >>> values = (1, 2, 3, 4, 5, 6, 7, 8)
+    >>> metadata._make(values)
+    DimCoordMetadata(standard_name=1, long_name=2, var_name=3, units=4, attributes=5, coord_system=6, climatological=7, circular=8)
+
+Note that, `namedtuple._make`_ is a class method, and so it is possible to
+create a **new** instance directly from the metadata class itself,
+
+    >>> from iris.common import DimCoordMetadata
+    >>> DimCoordMetadata._make(values)
+    DimCoordMetadata(standard_name=1, long_name=2, var_name=3, units=4, attributes=5, coord_system=6, climatological=7, circular=8)
+
+It is also possible to easily convert ``metadata`` to an `OrderedDict`_
+using the `namedtuple._asdict`_ method. This can be particularly handy when a
+standard Python built-in container is required to represent your ``metadata``,
+
+    >>> metadata._asdict()
+    OrderedDict([('standard_name', 'longitude'), ('long_name', None), ('var_name', 'longitude'), ('units', Unit('degrees')), ('attributes', {'grinning face': '🙃'}), ('coord_system', GeogCS(6371229.0)), ('climatological', False), ('circular', False)])
+
+Using the `namedtuple._replace`_ method allows you to create a new metadata
+class instance, but replacing specified members with **new** associated values,
+
+    >>> metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={'grinning face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+    >>> metadata._replace(standard_name=None, units=None)
+    DimCoordMetadata(standard_name=None, long_name=None, var_name='longitude', units=None, attributes={'grinning face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Another very useful method from the `namedtuple`_ toolkit is `namedtuple._fields`_.
+This method returns a tuple of strings listing the ``metadata`` members, in a
+fixed order. This allows you to easily iterate over the metadata class members,
+for what ever purpose you may require, e.g.,
+
+    >>> metadata._fields
+    ('standard_name', 'long_name', 'var_name', 'units', 'attributes', 'coord_system', 'climatological', 'circular')
+
+    >>> tuple([getattr(metadata, member) for member in metadata._fields])
+    ('longitude', None, 'longitude', Unit('degrees'), {'grinning face': '🙃'}, GeogCS(6371229.0), False, False)
+
+    >>> tuple([getattr(metadata, member) for member in metadata._fields if member.endswith("name")])
+    ('longitude', None, 'longitude')
+
+Note that, `namedtuple._fields`_ is also a class method, so you don't need
+an instance to determine the members of a metadata class, e.g.,
+
+    >>> from iris.common import CubeMetadata
+    >>> CubeMetadata._fields
+    ('standard_name', 'long_name', 'var_name', 'units', 'attributes', 'cell_methods')
```

Comment:
Fair point, but I think a lot of users haven't been exposed to namedtuples.

Also, if they have, then this highlights that you can easily rip apart the metadata, and iterate over it easily, rather than create your own enumerated list of members.

So, I think it has value... but that's based on speculation - that's why docs are hard as you don't know the experience level of the readership. One persons dross is another persons gold.